Feat(aigw): v2 azure claude cli (#5948)
Kong/developer.konghq.com: app/_how-tos/ai-gateway/use-claude-code-with-ai-gateway-azure.md

  a: Install {{ site.claude_code }}, create an AI Model Provider for your Azure AI Foundry Claude deployment, add a policy to strip Anthropic-only request fields Azure doesn't support, create an AI Model that targets it, then point {{ site.claude_code }}'s `ANTHROPIC_BASE_URL` at your local {{site.ai_gateway}} endpoint so all LLM requests pass through the gateway for monitoring and control.

---

## Configure an AI Model Provider

Create an [AI Model Provider](/ai-gateway/entities/ai-model-provider/) entity to define your connection to Azure and store your authentication credentials:

```sh
kongctl apply -f - --auto-approve --pat "$KONNECT_TOKEN" <<EOF
_defaults:
  kongctl:
    namespace: ai-gateway-get-started

ai_gateways:
  - ref: ai-quickstart
    _external:
      selector:
        matchFields:
          name: "ai-quickstart"

ai_gateway_model_providers:
  - ref: azure-claude
    name: azure-claude
    ai_gateway: ai-quickstart
    type: anthropic
    config:
      auth:
        type: basic
        headers:
          - name: x-api-key
            value: !env AZURE_AI_FOUNDRY_TOKEN
EOF
```

In this example, we're setting up the AI Model Provider with:

>  * `type: anthropic`: Specifies that this provider speaks Anthropic's native Messages API format. Azure AI Foundry serves Claude models through this same native API, so don't use `type: azure` — that driver assumes an Azure OpenAI-shaped deployment path (`/openai/deployments/<id>`) that Foundry's Claude endpoint doesn't use.

Reviewer: ```suggestion
 * `type: anthropic`: Specifies that this provider speaks Anthropic's native Messages API format. Azure AI Foundry serves Claude models through this same native API, so don't use `type: azure`. That driver assumes an Azure OpenAI-shaped deployment path (`/openai/deployments/<id>`) that Foundry's Claude endpoint doesn't use.
```

 * `name: azure-claude`: A unique identifier that AI Models will reference to route requests through this provider.
 * `config.auth.headers[0].name: x-api-key`: Azure AI Foundry's native Anthropic endpoint expects the API key in the `x-api-key` header, not `api-key` (which is specific to Azure OpenAI resources).
 * `config.auth.headers[0].value: !env AZURE_AI_FOUNDRY_TOKEN`: Loads the API key from your environment at apply time so it is not embedded in the config.

## Create a Request Transformer AI Policy

> Create an [AI Policy](/ai-gateway/entities/ai-policy/) entity using [request transformer](/ai-gateway/policies/ai-request-transformer/) to remove extra headers that Azure does not support.

Reviewer: ```suggestion
Create an [AI Policy](/ai-gateway/entities/ai-policy/) entity using [request transformer](/ai-gateway/policies/ai-request-transformer/) to remove extra headers that Azure doesn't support.
```

```sh
kongctl apply -f - --auto-approve --pat "$KONNECT_TOKEN" <<EOF
_defaults:
  kongctl:
    namespace: ai-gateway-get-started

ai_gateways:
  - ref: ai-quickstart
    _external:
      selector:
        matchFields:
          name: "ai-quickstart"

ai_gateway_models:
  - ref: claude-code-azure-sonnet
    name: claude-code-azure-sonnet
    display_name: "claude-code-azure-sonnet"
    ai_gateway: ai-quickstart
    type: model
    enabled: true
    formats:
      - type: anthropic
    config:
      route:
        paths:
          - /
      model:
        name_header: true
    capabilities:
      - generate
    policies: [ !ref claude-code-compat#name ]
    targets:
      - name: claude-sonnet-4-6
        provider: azure-claude
        config:
          type: anthropic
          upstream_url: !env AZURE_AI_FOUNDRY_UPSTREAM_URL
EOF
```

{:.info}
> Replace `claude-sonnet-4-6` with the name of your own Claude deployment in Azure AI Foundry.

In this example, we're setting up the AI Model with:

* `type: model`: Specifies this is a synchronous model for request/response workloads.

> * `name`/`display_name: claude-code-azure-sonnet`: The identifier you pass to `claude --model` — {{ site.claude_code }} uses this, not the upstream target name, to select the model.

Reviewer: ```suggestion
* `name`/`display_name: claude-code-azure-sonnet`: The identifier you pass to `claude --model`. {{ site.claude_code }} uses this, not the upstream target name, to select the model.
```

* `formats: [type: anthropic]`: Declares that this model accepts requests in Anthropic-compatible format, matching what {{ site.claude_code }} sends natively.

> * `config.route.paths: [/]`: Configures the base path where this model's Routes are accessible.

Reviewer: ```suggestion
* `config.route.paths: [/]`: Configures the base path where this model's routes are accessible.
```

* `config.model.name_header: true`: Lets {{ site.claude_code }} select this model by sending its `name` in the request, instead of requiring a separate `alias`.
* `capabilities: [generate]`: Enables text generation. For a model using the `anthropic` format, `generate` creates a `/messages` endpoint matching Anthropic's native Messages API, so combined with your base path, clients send requests to `/v1/messages`.
* `policies`: Attaches the `claude-code-compat` policy created in the previous step, so its header and body transformations apply to every request sent through this model.
* `targets`: Specifies which upstream model to route requests to. `provider: azure-claude` references the AI Provider created earlier, and `name: claude-sonnet-4-6` must match the name of your Claude deployment in Azure AI Foundry.

> * `targets[0].config.upstream_url`: The base Azure AI Foundry endpoint from the prerequisites, ending at `/anthropic` — {{site.ai_gateway}} appends the rest of the Anthropic Messages API path automatically.

Reviewer: ```suggestion
* `targets[0].config.upstream_url`: The base Azure AI Foundry endpoint from the prerequisites, ending at `/anthropic`. {{site.ai_gateway}} appends the rest of the Anthropic Messages API path automatically.
```

## Verify traffic through {{site.ai_gateway}}

Now, we can start a {{ site.claude_code }} session that points it to the local {{site.ai_gateway}} endpoint:

```sh
ANTHROPIC_BASE_URL=http://localhost:8000/ claude --model 'claude-code-azure-sonnet'
```

{{ site.claude_code }} asks for permission before it runs tools or interacts with files:

```text
I'll need permission to work with your files.

This means I can:
- Read any file in this folder
- Create, edit, or delete files
- Run commands (like npm, git, tests, ls, rm)
- Use tools defined in .mcp.json

Learn more ( https://docs.claude.com/s/claude-code-security )

❯ 1. Yes, continue
2. No, exit
```
{:.no-copy-code}

Select **Yes, continue**. The session starts. Ask a simple question to confirm that requests reach {{site.ai_gateway}}.

```text
Tell me about Vienna Oribasius manuscript.
```

> {{ site.claude_code }} might prompt you approve its web search for answering the question. When you select **Yes**, {{ site.claude }} will produce a full-length response to your request:

Reviewer: ```suggestion
{{ site.claude_code }} might prompt you to approve its web search for answering the question. When you select **Yes**, {{ site.claude }} will produce a full-length response to your request:
```
